src/app/shared/services/student.service.ts

```diff
@@ -0,0 +1,145 @@
+import { Injectable, Signal, WritableSignal, computed, inject, signal } from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+
+import {  Observable, Subject, defer } from 'rxjs';
+import { tap, exhaustMap } from 'rxjs/operators';
+
+import { Firestore, collection, orderBy, query, addDoc, updateDoc, serverTimestamp, deleteDoc, doc, DocumentReference, DocumentData } from 'firebase/firestore';
+import { collectionData } from 'rxfire/firestore';
+
+
+import { FIRESTORE } from '../../app.config';
+import { Student } from '../interfaces/students';
+
+interface StudentState {
+  students: Student[];
+  error: string | null;
+  loading: boolean;
+  selectedStudent: Student | null;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class StudentService {
+  private fireStore: Firestore = inject(FIRESTORE);
+
+  private students$: Observable<Student[]> = this.getStudents();
+  public addStudent$ = new Subject<Partial<Student>>();
+  public deleteStudent$ = new Subject<string>();
+  public updateStudent$ = new Subject<Partial<Student>>();
+  public selectStudent$ = new Subject<Student>();
```

Comment:
Let's rename these subjects, names look more like functions. May be select**ed**StudentSubject ?
Let's also use $ sign for observables and Subject postfix for subjects